curator_app/src/views/agent_list.js

```diff
@@ -14,15 +14,50 @@
  * limitations under the License.
  * ----------------------------------------------------------------------------
  */
+
 'use strict'
 
 const m = require('mithril')
+const sortBy = require('lodash/sortBy')
+const Table = require('../components/tables.js')
+const api = require('../services/api')
 
 const AgentList = {
+  oninit (vnode) {
+    vnode.state.agents = []
+
+    const refresh = () => {
+      api.get('agents').then((agents) => {
+        vnode.state.agents = sortBy(agents, 'name')
+      })
+        .then(() => { vnode.state.refreshId = setTimeout(refresh, 2000) })
```

Comment:
Do you need live refreshing? The way I implemented it is a very bad pattern, and should not be imitated by you or your students.

Ideally this is a web socket. Only keep if you absolutely must have live-reloading and absolutely do not have time to do web sockets.

Reply:
Also, even if you keep it, 5 seconds or 10 seconds is probably more reasonable that 2.